### PR TITLE
Deprecated warnings in tests

### LINKED ISSUE #1434: Fix Deprecation warning in test suite
### Bug Report
The test suite needs to remove deprecated warnings

### PR DESCRIPTION
**Pull request by justingluck93**
Fixes #1434 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
Fixed deprecation warnings in the test files.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by NicoleYarroch** [comment]
Can you remove the `f` from `fit` in the `SDLMenuManagerSpec`. The `f` means that only that test case should run.

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLDisplayCapabilitiesSpec.m [hunk 1]
```diff
@@ -52,7 +52,7 @@
     });
     
     it(@"Should get correctly when initialized", ^ {
-        NSMutableDictionary* dict = @{SDLRPCParameterNameDisplayType:SDLDisplayTypeGen26DMA,
+        NSMutableDictionary* dict = [@{SDLRPCParameterNameDisplayType:SDLDisplayTypeGen26DMA,
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLDisplayCapabilitiesSpec.m]: Make a simple `NSDictionary` instead

Comment (justingluck93) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLDisplayCapabilitiesSpec.m]: fixed

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLDisplayCapabilitiesSpec.m [hunk 2]
```diff
@@ -52,7 +52,7 @@
     });
     
     it(@"Should get correctly when initialized", ^ {
-        NSMutableDictionary* dict = @{SDLRPCParameterNameDisplayType:SDLDisplayTypeGen26DMA,
+        NSDictionary* dict = [@{SDLRPCParameterNameDisplayType:SDLDisplayTypeGen26DMA,
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLDisplayCapabilitiesSpec.m]: If you change this to an NSDictionary, you can't / shouldn't `mutableCopy`. Remove all `mutableCopy`s from this dictionary.

**Pull request merged**